source/vibe/web/rest.d

```diff
@@ -433,18 +433,22 @@ class RestInterfaceClient(I) : I
 			auto query = appender!string();
 
 			foreach (idx, p; params) {
-				static if (!(idx % 2)) {
-					// Parameter name
-					static assert(is(typeof(p) == string), "Internal error ("~__FUNCTION__~"): Parameter name is not string.");
-					query.put('&');
-					filterURLEncode(query, p);
-				} else {
-					// Parameter value
-					query.put('=');
-					static if (is(typeof(p) == Json))
-						filterURLEncode(query, p.toString());
-					else // Note: CTFE triggers compiler bug here (think we are returning Json, not string).
-						filterURLEncode(query, toRestString(serializeToJson(p)));
+				// gets value corresponding to the current parameter
+				auto value = params[(idx / 2) * 2 + 1];
+				// generates condition for inclusion of this parameter
+				enum condition = isNullable!(typeof(value)) ? q{ !value.isNull } : q{ true };
+				if (mixin(condition))
+				{
+					static if (!(idx % 2)) {
+						// Parameter name
+						static assert(is(typeof(p) == string), "Internal error ("~__FUNCTION__~"): Parameter name is not string.");
+						query.put('&');
+						filterURLEncode(query, p);
+					} else {
+						// Parameter value
+						query.put('=');
+						filterURLEncode(query, toRestString(p));
+					}
```

Comment:
I'm not sure the indexing logic is what you expected: `idx / 2 * 2 + 1` => [0, 1, 2, 3, 4, 5] => [1, 3, 3, 5, 5, 5]
So you never reference an even parameter (which would be a bug), but you reference the fitfh parameter thrice in a row. That means the bug would exhibit with 3 or more query parameters.

The logic in itself is quite hard to grasp (but the code was already obscure). I wouldn't mind an explanation in the code which explains why the `if` wraps the `static if` and not the other way around, and/or a little example of the code generated.